solidity/contracts/PhasedEscrow.sol

```diff
@@ -66,8 +66,96 @@ contract CurveRewardsEscrowBeneficiary is Ownable {
         curveRewards = _curveRewards;
     }
 
-    function __escrowSentTokens(uint256 amount) external {
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
         token.approve(address(curveRewards), amount);
         curveRewards.notifyRewardAmount(amount);
     }
 }
+
+/// @dev Interface of recipient contract for approveAndCall pattern.
+interface IStakerRewards {
+    function receiveApproval(
+        address _from,
+        uint256 _value,
+        address _token,
+        bytes calldata _extraData
+    ) external;
+}
+
+/// @title StakerRewardsBeneficiary
+/// @notice An abstract beneficiary contract that can receive a withdrawal phase
+///         from a PhasedEscrow contract. Immediately stakes the received tokens
+///         on a designated rewards escrow beneficiary contract.
+contract StakerRewardsBeneficiary is Ownable {
+    IERC20 public token;
+    IStakerRewards public stakerRewards;
+
+    constructor(IERC20 _token, IStakerRewards _stakerRewards) public {
+        token = _token;
+        stakerRewards = _stakerRewards;
+    }
+
+    function __escrowSentTokens(uint256 amount) external onlyOwner {
+        bool success = token.approve(address(stakerRewards), amount);
+        require(success, "Rewards token staking has failed");
```

Comment:
Why *staking*? 🤔 We are not staking here. I'd say `Token transfer approval failed`.

I guess *staking* for Curve rewards makes sense, but here, I'd rather say about funding reward contracts.

Reply:
Correct, it is just transfer of tokens, not staking, we're not locking tokens here. Changed.